k8s/staging/runners/protected/windows/x86_64/v2/release.yaml

```
@@ -0,0 +1,186 @@
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: runner-x86-v2-prot-windows
+  namespace: gitlab
+spec:
+  interval: 10m
+  url: https://charts.gitlab.io
+
+---
+apiVersion: helm.toolkit.fluxcd.io/v2beta2
+kind: HelmRelease
+metadata:
+  name: runner-x86-v2-prot-windows
+  namespace: gitlab
+spec:
+  interval: 10m
+  chart:
+    spec:
+      chart: gitlab-runner
+      version: 0.62.1 # gitlab-runner@16.9.1
+      sourceRef:
+        kind: HelmRepository
+        name: runner-x86-v2-prot-windows
+
+  valuesFrom:
+    # See terraform/modules/sentry/sentry.tf
+    - kind: ConfigMap
+      name: gitlab-runner-sentry-config
+      valuesKey: values.yaml
+
+  values:
+    image:
+      image: gitlab-org/gitlab-runner
+      tag: alpine-v16.7.0
+    imagePullPolicy: IfNotPresent
+    replicas: 1
+
+    # TODO: change to production when ready
+    gitlabUrl: "https://gitlab.staging.spack.io/"
+    unregisterRunners: true
+    terminationGracePeriodSeconds: 21600 # six hours
+    concurrent: 20
+    checkInterval: 30
+
+    metrics:
+      enabled: true
+
+    rbac:
+      serviceAccountName: runner
+
+    runners:
+      config: |
+        [[runners]]
+          pre_build_script = """
+          Write-Output 'Executing Spack pre-build setup script'
+
+          $PY3 = $null
+          $pythonCmds = @('python3', 'python')
+
+          foreach ($cmd in $pythonCmds) {
+              if (Get-Command -ErrorAction SilentlyContinue $cmd) {
+                  $PY3 = Get-Command $cmd
+                  break
+              }
+          }
+
+          if (-not $PY3) {
+              Write-Output "Unable to find python3 executable"
+              exit 1
+          }
+
+          $scriptPath = Join-Path $PSScriptRoot 'pre_build.py'
+
+          $wc = New-Object System.Net.WebClient
+          $wc.DownloadFile('https://raw.githubusercontent.com/spack/spack-infrastructure/main/scripts/gitlab_runner_pre_build/pre_build.py', $scriptPath)
+
+          & $PY3 $scriptPath | Out-File -FilePath 'envvars' -Encoding utf8
+
+          Remove-Item $scriptPath -Force
+          Remove-Item 'envvars' -Force
+
+          $env:GITLAB_OIDC_TOKEN = $null
+          """
+
+          shell = "powershell"
+          executor = "kubernetes"
+          output_limit = 4096
+
+          # Ensure windows paths are used
+          [runners.feature_flags]
+            FF_USE_POWERSHELL_PATH_RESOLVER = true
+
+          [runners.kubernetes]
+            namespace = "pipeline"
+            service_account = "runner"
+            # Allow 30 minutes for the pod to be ready. This is necessary because
+            # the windows docker image can take a significantly long time to pull.
+            poll_timeout = 1800
+            privileged = false
+
+            # Image for windows 2022, runner helper v16.1.0
+            helper_image = "gitlab/gitlab-runner-helper:x86_64-v16.7.0-servercore21H2"
```

```suggestion
            # Image for windows 2022, runner helper v16.7.0
            helper_image = "gitlab/gitlab-runner-helper:x86_64-v16.7.0-servercore21H2"
```

Also, it seems that this and the runner-image are specifically pointing to version 16.7.0 instead of 16.9.1 (which seems to be the version pointed to by the helm release). Is there a reason for that?

---

Oh, I must have missed this when I rebased this branch after a gitlab upgrade.

---

I'll add a note about this to the gitlab upgrade runbook

---

Ok I fixed this and pushed it, but for some reason github isn't showing it on this PR page. But you can see it if you go here https://github.com/spack/spack-infrastructure/compare/main...karpenter-windows-runners